frontend/src/app/plans/_components/PlansPage.tsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+
+import { atom, useAtom } from "jotai";
+import { PlusIcon } from "lucide-react";
+import { useRouter } from "next/navigation";
+import { useEffect, useRef } from "react";
+import { toast } from "sonner";
+
+import { planFamily } from "@/atoms/planFamily";
+import { plansIds } from "@/atoms/plansIds";
+import { PlanItem } from "@/components/PlanItem";
+
+import type { PlanResponseDataType } from "../page";
+
+const plansAtom = atom(
+  (get) => get(plansIds).map((id) => get(planFamily(id))),
+  (get, set, values: Array<{ id: string }>) => {
+    set(plansIds, values);
+  },
+);
+export function PlansPage({
+  plans: onlinePlans,
+}: {
+  plans: PlanResponseDataType[];
+}) {
+  const [plans, setPlans] = useAtom(plansAtom);
+  const router = useRouter();
+  const firstTime = useRef(true);
+
+  const addNewPlan = () => {
+    const uuid = crypto.randomUUID();
+    const newPlan = {
+      id: uuid,
+    };
+
+    void window.umami?.track("Create plan", {
+      numberOfPlans: plans.length,
+    });
+
+    router.push(`/plans/edit/${newPlan.id}`);
+    setPlans([...plans, newPlan]);
+  };
+
+  const plansExistingLocallyAndDeletedOnline = plans.filter(
+    (plan) =>
+      plan.onlineId !== null &&
+      !onlinePlans.some((p) => p.id.toString() === plan.onlineId),
+  );
+
+  const handleDeleteDeletedPlans = () => {
+    firstTime.current = false;
+    setPlans(
+      plans.filter(
+        (plan) =>
+          !plansExistingLocallyAndDeletedOnline.some((p) => p.id === plan.id),
+      ),
+    );
+    toast.success("Usunięto plany, które usunąłeś na innym urządzeniu.", {
+      duration: 5000,
+    });
+  };
+
+  useEffect(() => {
+    if (firstTime.current && plansExistingLocallyAndDeletedOnline.length > 0) {
+      handleDeleteDeletedPlans();
+    }
+  }, [plansExistingLocallyAndDeletedOnline]);
```

Comment:
o no chyba Cię pogrzało, co to niby jest 😭

Reply:
?

Reply:
jak masz lepszy pomysl na usuwanie lokalnych planow jak usune je na innym urzadzeniu to mow, chetnie przyjme pomysly bo tez mi sie nie podoba to ale nie mam innego pomyslu